homeassistant/components/husqvarna_automower/number.py

```diff
@@ -102,3 +166,74 @@ async def async_set_native_value(self, value: float) -> None:
             raise HomeAssistantError(
                 f"Command couldn't be sent to the command queue: {exception}"
             ) from exception
+
+
+class AutomowerWorkAreaNumberEntity(AutomowerBaseEntity, NumberEntity):
+    """Defining the AutomowerWorkAreaNumberEntity with AutomowerWorkAreaNumberEntityDescription."""
+
+    entity_description: AutomowerWorkAreaNumberEntityDescription
+
+    def __init__(
+        self,
+        mower_id: str,
+        coordinator: AutomowerDataUpdateCoordinator,
+        description: AutomowerWorkAreaNumberEntityDescription,
+        work_area_id: int,
+    ) -> None:
+        """Set up AutomowerNumberEntity."""
+        super().__init__(mower_id, coordinator)
+        self.entity_description = description
+        self.work_area_id = work_area_id
+        self._attr_unique_id = f"{mower_id}_{work_area_id}_{description.key}"
+        self._attr_translation_placeholders = {"work_area": self.work_area.name}
+
+    @property
+    def work_area(self) -> WorkArea:
+        """Get the mower attributes of the current mower."""
+        if TYPE_CHECKING:
+            assert self.mower_attributes.work_areas is not None
+        return self.mower_attributes.work_areas[self.work_area_id]
+
+    @property
+    def translation_key(self) -> str:
+        """Return the translation key of the work area."""
+        return self.entity_description.translation_key_fn(self.work_area_id)
+
+    @property
+    def native_value(self) -> float:
+        """Return the state of the number."""
+        return self.entity_description.value_fn(self.work_area)
+
+    async def async_set_native_value(self, value: float) -> None:
+        """Change to new number value."""
+        try:
+            await self.entity_description.set_value_fn(
+                self.coordinator, self.mower_id, value, self.work_area_id
+            )
+        except ApiException as exception:
+            raise HomeAssistantError(
+                f"Command couldn't be sent to the command queue: {exception}"
+            ) from exception
+
+
+async def async_remove_entities(
+    coordinator: AutomowerDataUpdateCoordinator,
+    hass: HomeAssistant,
```

Comment:
Always start with the `hass` parameter in the signature if it exists in the signature.